homeassistant/components/wolflink/sensor.py

```diff
@@ -0,0 +1,162 @@
+"""The Wolf SmartSet sensors."""
+import logging
+
+from wolf_smartset.models import (
+    HoursParameter,
+    ListItemParameter,
+    Parameter,
+    PercentageParameter,
+    Pressure,
+    SimpleParameter,
+    Temperature,
+)
+
+from homeassistant.components.wolflink.const import DOMAIN, STATES
+from homeassistant.const import (
+    DEVICE_CLASS_PRESSURE,
+    DEVICE_CLASS_TEMPERATURE,
+    PRESSURE_BAR,
+    TEMP_CELSIUS,
+    TIME_HOURS,
+)
+from homeassistant.helpers.entity import Entity
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(hass, config_entry, async_add_entities):
+    """Set up all entries for Wolf Platform."""
+
+    data = hass.data[DOMAIN][config_entry.entry_id]["data"]
+
+    parameters = data.parameters
+
+    entities = []
+    for parameter in parameters:
+        if isinstance(parameter, Temperature):
+            entities.append(WolfLinkTemperature(data, parameter))
+        if isinstance(parameter, Pressure):
+            entities.append(WolfLinkPressure(data, parameter))
+        if isinstance(parameter, PercentageParameter):
+            entities.append(WolfLinkPercentage(data, parameter))
+        if isinstance(parameter, ListItemParameter):
+            entities.append(WolfLinkState(data, parameter))
+        if isinstance(parameter, HoursParameter):
+            entities.append(WolfLinkHours(data, parameter))
+        if isinstance(parameter, SimpleParameter):
+            entities.append(WolfLinkSensor(data, parameter))
+
+    async_add_entities(entities, True)
+
+
+class WolfLinkSensor(Entity):
+    """Base class for all Wolf entities."""
+
+    def __init__(self, wolflink_data, wolf_object: Parameter):
+        """Initialize."""
+        self.wolflink_data = wolflink_data
+        self.wolf_object = wolf_object
+        self._state = None
+
+    @property
+    def name(self):
+        """Return the name."""
+        return f"{self.wolf_object.name}"
+
+    @property
+    def state(self):
+        """Return the state."""
+        return self._state
+
+    @property
+    def device_state_attributes(self):
+        """Return the state attributes."""
+        return {
+            "parameter_id": self.wolf_object.parameter_id,
+            "value_id": self.wolf_object.value_id,
+            "parent": self.wolf_object.parent,
```

Comment:
Huh. It is tough question. Since this stupid API is returning tree of parameters and sometimes there is the same parameter name and I would like to have possibility to tell if entity called 'Zeitprogramme' is for Heating or Warm water. And yes I think that this will be very useful during automations.

Reply:
Ok.